src/lib/request.js

```diff
@@ -71,8 +71,10 @@ const request = async ({
                 } else if (responseType === ResponseType.JSON) {
                     const contentType = res.headers['content-type'];
                     if (!/^application\/json/.test(contentType)) {
-                        return reject(new Error('Invalid content-type. ' +
-                            `Expected application/json but received ${contentType}`));
+                        let err = new Error('Invalid content-type. ' +
```

Comment:
I would use `const` - even if you modify some props

Reply:
I disagree, in my opinion `const` is best used as a contract with the reader: "I will not change this data". Obviously js doesn't enforce this, unfortunately, but that doesn't mean we can't use it as such.

Reply:
I tend to avoid const if I know some props will be changed, as Matt says, if you are going to change it, dont call it const.

Reply:
**my 3 cents** about `const` misconceptions:
You don't modify `err` so it is `const` in javascript word, where you still can modify its properties which aren't const by language design.

There is a difference between `const` in c++ which has a `deep` semantic that is opposite to the `shallow` concept of `const`  in Javascript which protects only against reassigning of the value for simple types and the reference for the complex ones.

Out of scope: If you want to protect an object's properties against reassignment you can use `Object.freeze()`